libraries/botbuilder/src/teamsActivityHandler.ts

```diff
@@ -606,6 +606,88 @@ export class TeamsActivityHandler extends ActivityHandler {
         }
     }
 
+    /**
+     * Override this method to change the dispatching of MessageUpdate activities.
+     *
+     * @param context A context object for this turn.
+     * @returns A promise that represents the work queued.
+     */
+    protected async dispatchMessageUpdateActivity(context: TurnContext): Promise<void> {
+        if (context.activity.channelId == 'msteams') {
+            const channelData = context.activity.channelData as TeamsChannelData;
+
+            switch (channelData.eventType) {
+                case 'undeleteMessage':
+                    return await this.onTeamsMessageUndelete(context);
+
+                case 'editMessage':
+                    return await this.onTeamsMessageEdit(context);
```

Comment:
Since this is in Teams, which scenarios does this support?

Off the top of my head I have the following scenarios:

1. 1:1 conversations
2. when the bot receives all activities inside of the Team its installed in
3. if a user edits a message and adds an @mention addressed to the bot
4. if a user edits a message and removes an @mention addressed to the bot

What's the end scenarios look like?

Reply:
These have been discussed and vetted already via our Teams Sync and review meetings.  Is there a reason you are asking?